### DISCUSSION TITLE
How to open the QFile file picker by pressing a QBtn?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#13641] [General - Components / Directives / etc] [answered]
I would like to replace the generic input of QFile with a QBtn - pressing the QBtn would trigger the opening of the native OS file picker (as this is the case today when I click on the input field).

Is there a way to do that with Quasar or do I have to use `window.showOpenFilePicker()`?

**Comment by MilosPaunovic** [comment, marked as answer]
Use `refs` and `pickFiles` method.

**Reply by wsw70** [reply]
How does it work, exactly? If I add a `ref=...` to the attributes of `<q-file>` then I will see an input form anyay, right? (I understand that from there I can somehow use the `ref` to access the proper method as part as a `@click` of a QBtn, but the native QFile input will still be there)

**Reply by suleimanas** [reply]
you can hide it and still use it
